File: python/java_to_python/04-pgmContador/pgmContador.py
import datetime
import time

class PgmContador:
    def __init__(self):
        print('|----------------------------------|')
        print('|-------------PGM: printf----------|')
        print('|---Author:🐍 Santiago Duran 🐍----|')
        print('|----------------------------------|')
        
    def contar(self):
        try:
            print("Contando desde el shell a las "+str(datetime.datetime.now()))
            for i in range(20):
                print("Posicion Indice ❇️: "+str(i))
                time.sleep(1-(i*0.05))
            print("Contador Finalizado...🔥🔥🔥🔥🔥")
        except Exception as e:
            print("⚠️⚠️⚠️ "+str(e))

pgm = PgmContador()
pgm.contar()

File: ruby/pgms_ruby/04-pgmContador/pgmContador.rb
require 'date'

class PgmContador
  def initialize
    puts '|----------------------------------|'
    puts '|-------------PGM: printf----------|'
    puts '|---Author:🐍 Santiago Duran 🐍----|'
    puts '|----------------------------------|'
  end
  
  def contar
    begin
      puts "Contando desde el shell a las #{Time.now}"
      (0..19).each do |i|
        puts "Posicion Indice ❇️: #{i}"
        sleep(1 - (i * 0.05))
      end
      puts "Contador Finalizado...🔥🔥🔥🔥🔥"
    rescue => e
      puts "⚠️⚠️⚠️ #{e}"
    end
  end
end

pgm = PgmContador.new
pgm.contar